Calibration/TkAlCaRecoProducers/plugins/AlcaBeamSpotHarvester.cc

```diff
@@ -155,7 +157,7 @@ void AlcaBeamSpotHarvester::endRun(const edm::Run& iRun, const edm::EventSetup&)
         //poolDbService->appendSinceTime<BeamSpotObjects>(aBeamSpot, poolDbService->currentTime(),"BeamSpotObjectsRcd");
 	poolDbService->writeOne<BeamSpotObjects>(aBeamSpot, thisIOV, outputrecordName_);
         if (dumpTxt_ && beamSpotOutputBase_ == "lumibased"){
-            beamspot::dumpBeamSpotTxt(outTxt, true, currentBS);
+            beamspot::dumpBeamSpotTxt(outFile, currentBS);
```

Comment:
the origin of the problem comes from duplicating the file handling in the printer function, here
https://github.com/cms-sw/cmssw/blob/master/RecoVertex/BeamSpotProducer/interface/BeamSpotWrite2Txt.h#L22-L27

while this is already done here
https://github.com/cms-sw/cmssw/blob/master/RecoVertex/BeamSpotProducer/src/BeamFitter.cc#L636-L639

==============

As long as the outfile is used in append mode, everything is normal.

The problem showed up because, when the append option is not enabled, and this only happens for DIP, the outfile is first open here
https://github.com/cms-sw/cmssw/blob/master/RecoVertex/BeamSpotProducer/src/BeamFitter.cc#L636-L639

then reopened by the printer function here
https://github.com/cms-sw/cmssw/blob/master/RecoVertex/BeamSpotProducer/interface/BeamSpotWrite2Txt.h#L22-L27
the beam spot fit results are written, and then the file is closed here
https://github.com/cms-sw/cmssw/blob/master/RecoVertex/BeamSpotProducer/interface/BeamSpotWrite2Txt.h#L55
but then, back in the main BeamFitter.cc file, there's a concurrent attempt writing into the same file here
https://github.com/cms-sw/cmssw/blob/master/RecoVertex/BeamSpotProducer/src/BeamFitter.cc#L690-L701


Now, the idea is that the printer function does not need to handle the creation/closure and write mode of the outfile and let the code that calls it do it.